tests/components/axis/test_device.py

```diff
@@ -198,30 +187,82 @@
 """
 
 
-async def vapix_request(self, session, url, **kwargs):
-    """Return data based on url."""
-    if API_DISCOVERY_URL in url:
-        return API_DISCOVERY_RESPONSE
-    if APPLICATIONS_URL in url:
-        return APPLICATIONS_LIST_RESPONSE
-    if BASIC_DEVICE_INFO_URL in url:
-        return BASIC_DEVICE_INFO_RESPONSE
-    if LIGHT_CONTROL_URL in url:
-        return LIGHT_CONTROL_RESPONSE
-    if MQTT_CLIENT_URL in url:
-        return MQTT_CLIENT_RESPONSE
-    if PORT_MANAGEMENT_URL in url:
-        return PORT_MANAGEMENT_RESPONSE
-    if VMD4_URL in url:
-        return VMD4_RESPONSE
-    if BRAND_URL in url:
-        return BRAND_RESPONSE
-    if IOPORT_URL in url or INPUT_URL in url or OUTPUT_URL in url:
-        return PORTS_RESPONSE
-    if PROPERTIES_URL in url:
-        return PROPERTIES_RESPONSE
-    if STREAM_PROFILES_URL in url:
-        return STREAM_PROFILES_RESPONSE
+def mock_default_vapix_requests(respx: respx, host: str = "1.2.3.4") -> None:
+    """Mock default Vapix requests responses."""
+    respx.post(f"http://{host}:80/axis-cgi/apidiscovery.cgi").respond(
+        json=API_DISCOVERY_RESPONSE,
+        headers={"Content-Type": "application/json"},
```

Comment:
We get automatic content type when specifying `json` parameter. Is that enough?

```
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.18.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import httpx, respx

In [2]: def test_ctx_manager():
   ...:     with respx.mock:
   ...:         my_route = respx.post("https://example.org/").respond(
   ...:             json={"test": "one"}
   ...:         )
   ...:         response = httpx.post("https://example.org/")
   ...:         print(response.headers)
   ...: 

In [3]: test_ctx_manager()
Headers({'content-length': '15', 'content-type': 'application/json'})
```

Reply:
Absolutely